docs/static/keystore.asciidoc

```diff
@@ -0,0 +1,152 @@
+`${KEY}`
+
+Where KEY is the name of the key.
+
+For example, imagine that the keystore contains a key called `ES_PWD` with the
+value `yourelasticsearchpassword`:
+
+* In the configuration file, use `output {  elasticsearch {  ...  password => "${ES_PWD}" } } }`
+* In `logstash.yml`, use: `xpack.management.elasticsearch.password: ${ES_PWD}`
+
+NOTE: Referencing keystore data from `pipelines.yml` or the command line (`-e`)
+is not currently supported.
+
+When Logstash parses the settings (`logstash.yml`) or configuration
+(`/etc/logstash/conf.d/*.conf`), it resolves keys from the keystore before
+resolving environment variables.
+
+// TODO: add keystore-command to running-logstash-command-line.asciidoc
+// To create and manage keys, use the `keystore` command. See the
+// <<keystore-command,command reference>> for the full command syntax, including
+// optional flags.
+
+[float]
+[[keystore-password]]
+=== Keystore password
+
+You can protect access to the Logstash keystore by storing a password in an
+environment variable called `LOGSTASH_KEYSTORE_PASS`. If you create the Logstash
+keystore after setting this variable, the keystore will be password protected.
+This means that the environment variable needs to be accessible to the running
+instance of Logstash. This environment variable must also be correctly set for
+any users who need to issue keystore commands (add, list, remove, etc.).
+
+Using a keystore password is recommended, but optional. The data will be encrypted even if you
+do not set a password. However, it is highly recommended to configure the
+keystore password and grant restrictive permissions to any files that may
+contain the environment variable value.
+
+For example:
+[source,sh]
+--------------------------------------------------
+set +o history
+export LOGSTASH_KEYSTORE_PASS=mypassword
+bin/logstash-keystore create
+--------------------------------------------------
+
+This setup requires the user running Logstash to have the environment variable
+`LOGSTASH_KEYSTORE_PASS=mypassword` defined. If it is not defined, Logstash
+cannot access the the keystore.
+
+When you run Logstash from an RPM or DEB package installation, the environment
+variables are sourced from `/etc/sysconfig/logstash`.
+
+NOTE: You might need to create `/etc/sysconfig/logstash`. This file should be
+owned by `root` with `600` permissions. The expected format of
+`/etc/sysconfig/logstash` is `ENVIRONMENT_VARIABLE=VALUE`, with one entry per
+line.
+
+For other distributions, such as Docker or ZIP, see the documentation for your
+runtime environment (Windows, Docker, etc) to learn how to set the
+environment variable for the user that runs Logstash. Ensure that the
+environment variable (and thus the password) is only accessible to that user.
+
+[float]
+[[keystore-location]]
+=== Keystore location
+
+The keystore must be located in Logstash's `path.settings` directory. This is
+the same directory that contains the `logstash.yml` file. When performing any
```

Comment:
isn't the logstash.yml in LS_HOME/config?

Reply:
Specifically logstash.yml is found in `path.settings` location, and `path.settings` is defaulted (in the source code) to LS_HOME/config.  

However, the RPM/DEB installations [override](https://github.com/elastic/logstash/blob/master/pkg/startup.options) this to `etc/logstash`  https://www.elastic.co/guide/en/logstash/current/dir-layout.html has the complete listing of directories.